src/sentry/static/sentry/app/components/organizations/globalSelectionHeader/utils.jsx

```diff
@@ -2,17 +2,20 @@ import {pick, pickBy, identity} from 'lodash';
 
 import {defined} from 'app/utils';
 import {getUtcToLocalDateObject} from 'app/utils/dates';
+import {getParams} from 'app/views/events/utils/getParams';
 
 import {URL_PARAM} from 'app/constants/globalSelectionHeader';
 
 // Parses URL query parameters for values relevant to global selection header
 export function getStateFromQuery(query) {
-  let start = query[URL_PARAM.START] !== 'null' && query[URL_PARAM.START];
-  let end = query[URL_PARAM.END] !== 'null' && query[URL_PARAM.END];
+  const parsedParams = getParams(query);
```

Comment:
I'm seeing some issues stemming from the global selection header propagating down to discover2.

Invalid  datetime selections can easily throw an error and break the app with no actual recovery flow.

I feel it's better to parse and validate the datetime selection early here; rather than doing it at the API payload generation everywhere.